sqlglot/parser.py

```diff
@@ -2907,6 +2907,9 @@ def _parse_join(
         if hint:
             kwargs["hint"] = hint
 
+        if self._match(TokenType.MATCH_CONDITION):
```

Comment:
Can we just `_match_text_seq` here? We've been avoiding introducing new tokens.

Reply:
That was my initial implementation, but turns out that `a ASOF JOIN b MATCH_CONDITION(...)` will result in `MATCH_CONDITION` being parsed as `b`'s TableAlias, so I thought the proper way forward would be to declare a token in order to impose finer control. Are there any easy workarounds to this issue that I'm missing?

Reply:
I see. I believe that the current implementation will also result in the behavior described in https://github.com/tobymao/sqlglot/issues/3247#issuecomment-2028481995.

Basically, `TABLE_ALIAS_TOKENS` won't contain the newly added token even though it's contained in `ID_VAR_TOKENS`. This is a side-effect and a non-trivial one at that. It may change in the future if we decide to create the former constant in the metaclass.

Maybe we can keep what you have here for now. Perhaps do this as well to make it more future-proof:

```python
TABLE_ALIAS_TOKENS.pop(TokenType.MATCH_CONDITION, None)
```

We can discuss later about how we want to deal with the linked issue.